services/auth/middleware.go

```diff
@@ -197,7 +211,7 @@ func VerifyAuthWithOptionsAPI(options *VerifyOptions) func(ctx *context.APIConte
 				return
 			} else if !ctx.Doer.IsActive && setting.Service.RegisterEmailConfirm {
 				ctx.Data["Title"] = ctx.Tr("auth.active_your_account")
-				ctx.HTML(http.StatusOK, "user/auth/activate")
+				ctx.JSON(http.StatusOK, map[string]any{"ok": true})
 				return
```

Comment:
Because, this is a API middleware, it doesn't make sense to return a HTML page.

The old code was incorrect either, the code was just copied&pasted, it's a legacy problem and no good solution yet.

Reply:
IMO  the  best it  could be is  like this : aa4bd9aa4a8e8ac0a6e1abd0ca7c4641ee78f964 , at least, the  same logic is also at line 177 .  Further changes  could   be  done in  future  PRs.